### ISSUE TITLE
How to set the batch size of a training startegy instance? 

### ISSUE THREAD

**Issue by leondavi** [Artelnics/opennn#224]
Two questions related to opennn training startegy: 
1. How to get the value of the training error? 
2. How to set the batch size of the training strategy instance?. When running the training strategy there is a printing of "Batch size: 1000" but I couldn't find how can I control this parameter.

**Comment by sandstromviktor** [comment]
One way of doing question 1 is as follows:

 ```
        TrainingStrategy training_strategy(&neural_network, &data_set);

        // Specify the optimization method to use. 
        training_strategy.set_optimization_method(TrainingStrategy::OptimizationMethod::ADAPTIVE_MOMENT_ESTIMATION);

        // If we want to change the default values, we must create a pointer to that specific optimizer method 
        AdaptiveMomentEstimation* cool_pointer = training_strategy.get_adaptive_moment_estimation_pointer();

        // Print the new learning rate, should be 0.001
        cout << "Old learning rate: " << cool_pointer->get_initial_learning_rate() << endl;

        cool_pointer->set_initial_learning_rate(0.0005);

        // Print the new learning rate 
        cout << "New learning rate: " << cool_pointer->get_initial_learning_rate() << endl;

```

**Comment by sandstromviktor** [comment]
At what point in the training do you want the training error?
At the end of training you can do
`        TrainingResults results = training_strategy.perform_training();
        cout << results.get_training_error() << endl;`

**Comment by leondavi** [comment]
> At what point in the training do you want the training error? At the end of training you can do ` TrainingResults results = training_strategy.perform_training(); cout << results.get_training_error() << endl;`

I am looking for a vector of the training error that shows the error change over the history of training (for example how the error is changed for each sample)

**Comment by sandstromviktor** [comment]
There is a function `results.training_error_history()`. Don't know if this is for each batch or for each epoch (probably the latter). 
For each sample is probably more difficult since you calculate the loss for the entire batch (assuming batch_size > 1). 
This is why i enjoy the PyTorch-API where you specify everything inside your custom training loop. For standard cases this works fine, but i really hope the developers go the PyTorch-way eventually.
